int64 entryPoint
int32[] robots